central/policy/service/service_impl.go

```diff
@@ -1122,3 +1125,52 @@ func combineStrings(toCombine [][]string) []string {
 	}
 	return nil
 }
+
+func (s *serviceImpl) SaveAsCustomResources(ctx context.Context, request *v1.SaveAsCustomResourcesRequest) (*v1.SaveAsCustomResourcesResponse, error) {
+	policyList, missingIndices, err := s.policies.GetPolicies(ctx, request.GetPolicyIds())
+	if err != nil {
+		return nil, err
+	}
+	errDetails := &v1.PolicyOperationErrorList{}
+	for _, missingIndex := range missingIndices {
+		policyID := request.GetPolicyIds()[missingIndex]
+		errDetails.Errors = append(errDetails.Errors, &v1.PolicyOperationError{
+			PolicyId: policyID,
+			Error: &v1.PolicyError{
+				Error: "not found",
+			},
+		})
+		log.Warnf("A policy error occurred for id %s: not found", policyID)
+	}
+	if len(missingIndices) > 0 {
+		statusMsg, err := status.New(codes.InvalidArgument, "Some policies could not be retrieved. Check the error details for a list of policies that could not be found").WithDetails(errDetails)
```

Comment:
Failed to retrieve all policies. Check error details for a list of policies that could not be retrieved.